src/main/java/com/PBA/authservice/persistance/repository/JdbcRepository.java

```diff
@@ -0,0 +1,76 @@
+package com.pba.authservice.persistance.repository;
+
+import com.pba.authservice.exceptions.AuthDaoException;
+import org.springframework.jdbc.core.JdbcTemplate;
+import com.pba.authservice.persistance.repository.sql.SqlProvider;
+import org.springframework.jdbc.core.RowMapper;
+
+import java.lang.reflect.Field;
+import java.util.*;
+
+public abstract class JdbcRepository<ObjectT, IdT> {
+    private final RowMapper<ObjectT> rowMapper;
+    private final SqlProvider sqlProvider;
+    private final JdbcTemplate jdbcTemplate;
+
+    public JdbcRepository(RowMapper<ObjectT> rowMapper, SqlProvider sqlProvider, JdbcTemplate jdbcTemplate) {
+        this.rowMapper = rowMapper;
+        this.sqlProvider = sqlProvider;
+        this.jdbcTemplate = jdbcTemplate;
+    }
+
+    private List<Object> extractAttributes(Object obj) {
+        List<Field> fields = Arrays.stream(obj.getClass().getDeclaredFields()).toList();
+        fields.forEach((field) -> field.setAccessible(true));
+        List<Object> attributes = fields.stream()
+                    .map((field) -> {
+                            try {
+                                return field.get(obj);
+                            } catch (IllegalAccessException e) {
+                                throw new RuntimeException(e);
+                            }
+                    }).toList();
+        List<Object> mutableAttributes = new ArrayList<>(attributes);
```

Comment:
Instead of using this approach, use skip(1) on stream above: https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html#skip-long-

but why do you have to skip the id?